integration/nevermined/SecondaryMarket.test.ts

```diff
@@ -0,0 +1,713 @@
+                    agreementAccessId2,
+                    ddo,
+                    new AssetRewards(),
+                    collector2,
+                    numberNFTs,
+                    collector2
+                )
+                assert.isTrue(result)
+
+                const status = await nftAccessTemplate.getAgreementStatus(
+                    agreementAccessId2
+                )
+                assert.equal(status && status.nftHolder.state, ConditionState.Unfulfilled)
+                assert.equal(status && status.nftAccess.state, ConditionState.Unfulfilled)
+            })
+
+            it('The collector2 demonstrates it onws the NFT', async function() {
+                // See https://github.com/nevermined-io/sdk-js/issues/137
+                if (networkName === 'polygon-localnet') {
+                    this.skip()
+                }
+
+                // TODO: Not sure why we need to wait here but without this the
+                // the fulfillment will fail
+                await new Promise(r => setTimeout(r, 10000))
+                const result = await nevermined.agreements.conditions.holderNft(
+                    agreementAccessId2,
+                    ddo.id,
+                    collector2.getId(),
+                    numberNFTs
+                )
+                assert.isTrue(result)
+            })
+
+            it('The artist gives access to the collector2 to the content', async function() {
+                // See https://github.com/nevermined-io/sdk-js/issues/137
+                if (networkName === 'polygon-localnet') {
+                    this.skip()
+                }
+
+                const result = await nevermined.agreements.conditions.grantNftAccess(
+                    agreementAccessId2,
+                    ddo.id,
+                    collector2.getId(),
+                    artist
+                )
+                assert.isTrue(result)
+            })
+
+            it('Collector 1 no longer has access the to the content', async () => {
+                // Not the best way to do this but on spree we don't get the revert reasons
+                await assert.isRejected(
+                    nevermined.agreements.conditions.holderNft(
+                        agreementAccessId,
+                        ddo.id,
+                        collector1.getId(),
+                        numberNFTs
+                    )
+                )
+            })
+
+            it('As collector2 I setup an agreement for selling my NFT', async () => {
+                agreementId3 = await nevermined.nfts.listOnSecondaryMarkets(
+                    ddo,
+                    assetRewards3,
+                    numberNFTs2,
+                    collector2.getId(),
+                    token,
+                    collector2.getId()
+                )
+                assert.isNotNull(agreementId3)
+
+                const service = await nevermined.metadata.retrieveService(agreementId3)
```

Comment:
the tests fail because service is undefined

Reply:
Yeah, i saw that, it was just that locally it had no problem in retrieving it... ❓